tests/phpunit/tests/cron.php

```diff
@@ -1022,6 +1025,48 @@ public function test_schedule_short_circuit_with_false_returns_error_when_wp_err
 		$this->assertSame( 'pre_reschedule_event_false', $rescheduled_event->get_error_code() );
 	}
 
+	/**
+	 * @ticket 57271
+	 *
+	 * @dataProvider data_wp_reschedule_event_args
+	 *
+	 * @covers ::wp_reschedule_event
+	 */
+	public function test_wp_reschedule_event_works_with_args( array $args ) {
+		$time = time();
+
+		// Schedule events with the `$wp_error` parameter:
+		$event             = wp_schedule_event( $time, 'daily', 'hook', $args, true );
+		$rescheduled_event = wp_reschedule_event( $time, 'daily', 'hook', $args, true );
+		$unscheduled_event = wp_unschedule_event( $time, 'hook', $args, true );
+		$next_timestamp    = wp_next_scheduled( 'hook', $args );
+
+		// Ensure the events were added and updated correctly:
+		$this->assertNotWPError( $event );
+		$this->assertNotWPError( $rescheduled_event );
+		$this->assertNotWPError( $unscheduled_event );
+		$this->assertSame( $time + DAY_IN_SECONDS, $next_timestamp );
+	}
+
+	public function data_wp_reschedule_event_args() {
```

Comment:
Docblock & renaming.

```suggestion
	/**
	 * Data provider for test_wp_reschedule_event_works_with_args().
	 *
	 * @return array[]
	 */
	public function data_wp_reschedule_event_works_with_args() {
```